src/Compilers/VisualBasic/Portable/Binding/Binder_Conversions.vb

```diff
@@ -1644,6 +1644,11 @@ DoneWithDiagnostics:
                 ' do not lose the original element names in the literal if different from names in the target
                 ' Come back to this, what about locations? (https:'github.com/dotnet/roslyn/issues/11013)
                 targetType = destTupleType.WithElementNames(sourceTuple.ArgumentNamesOpt)
+
+                If Not destTupleType.AllRequiredMembers.IsEmpty Then
```

Comment:
>If Not destTupleType.AllRequiredMembers.IsEmpty Then [](http://example.com/codeflow?start=16&length=52)

The approach of intercepting tuple types during binding feels very fragile. I think it would be better to mark the type symbol as bad via ```GetUseSiteInfo``` API. While we are calculating the diagnostics, we probably can check if the type has the "tuple" name and arity between 1 and 8, and add ```ERR_UnsupportedType1`` if it has required members. We are not calculating use-site errors for types declared in source, but that shouldn't be a problem because VB cannot declare a type with required members in source.